packages/react/src/migrations/update-9-4-0/babelrc-9-4-0.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  chain,
+  Rule,
+  SchematicContext,
+  Tree,
+} from '@angular-devkit/schematics';
+import { getFullProjectGraphFromHost } from '@nrwl/workspace/src/utils/ast-utils';
+import { stripIndent } from '@angular-devkit/core/src/utils/literals';
+import { initRootBabelConfig } from '@nrwl/web/src/utils/rules';
+
+/*
+ * This migration will do a few things:
+ *
+ * - Create the base babel.config.json file if it doesn't already exist
+ * - Create .babelrc files for each React project that doesn't already have one
+ * - For any projects that are not migrated, display a message so users are not surprised.
+ */
+export default function update(): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    const updates = [];
+    const conflicts: Array<[string, string]> = [];
+    const projectGraph = getFullProjectGraphFromHost(host);
+    if (host.exists('/babel.config.json')) {
+      context.logger.info(
+        `
+        Found an existing babel.config.json file so we skipped creating it.
+        
+        You may want to update it to include the Nx preset "@nrwl/web/babel".
+        `
+      );
+    } else if (host.exists('/babel.config.js')) {
+      context.logger.info(
+        `
+        Found an existing babel.config.js file so we skipped creating it.
+        
+        You may want to update it to include the Nx preset "@nrwl/web/babel".
+        `
+      );
+    } else {
+      updates.push(initRootBabelConfig());
+    }
+
+    Object.keys(projectGraph.nodes).forEach((name) => {
+      const p = projectGraph.nodes[name];
+      const deps = projectGraph.dependencies[name];
+      const isReact = deps.some(
```

Comment:
Hm... isn't it cheaper to go through projects with the `@nrwl/web:build` builder?

Getting the whole dep-graph (with no cache) is overkill to find projects that depend on react?

Reply:
Nevermind. Makes sense now! :+1: